ignoring return value of ‘virtual bool QFile::open(QIODeviceBase::OpenMode)’

### Operating System

_No response_

### Version of Prism Launcher

develop

### Version of Qt

6.10.0

### Description of bug

when building prism the cmake complains about the no discard attribute on openFile

### Steps to reproduce

build prism

### Suspected cause

```
[129/558] Building CXX object launcher/CMakeFiles/Launcher_logic.dir/FileSystem.cpp.o
.../PrismLauncher/launcher/FileSystem.cpp: In function ‘QString FS::createShortcut(QString, QString, QStringList, QString, QString)’:
.../PrismLauncher/launcher/FileSystem.cpp:1005:11: warning: ignoring return value of ‘virtual bool QFile::open(QIODeviceBase::OpenMode)’, declared with attribute ‘nodiscard’ [-Wunused-result]
 1005 |     f.open(QIODevice::WriteOnly | QIODevice::Text);
      |     ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/qt6/QtCore/qdir.h:11,
                 from /usr/include/qt6/QtCore/QDir:1,
                 from .../PrismLauncher/launcher/FileSystem.h:45,
                 from .../PrismLauncher/launcher/FileSystem.cpp:38:
/usr/include/qt6/QtCore/qfile.h:291:32: note: declared here
  291 |     QFILE_MAYBE_NODISCARD bool open(OpenMode flags) override;
      |                                ^~~~
[137/558] Building CXX object launcher/CMakeFiles/Launcher_logic.dir/net/HttpMetaCache.cpp.o
.../PrismLauncher/launcher/net/HttpMetaCache.cpp: In member function ‘MetaEntryPtr HttpMetaCache::resolveEntry(QString, QString, QString)’:
.../PrismLauncher/launcher/net/HttpMetaCache.cpp:115:19: warning: ignoring return value of ‘virtual bool QFile::open(QIODeviceBase::OpenMode)’, declared with attribute ‘nodiscard’ [-Wunused-result]
  115 |         input.open(QIODevice::ReadOnly);
      |         ~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/qt6/QtCore/qdir.h:11,
                 from /usr/include/qt6/QtCore/QDir:1,
                 from .../PrismLauncher/launcher/FileSystem.h:45,
                 from .../PrismLauncher/launcher/net/HttpMetaCache.cpp:37:
/usr/include/qt6/QtCore/qfile.h:291:32: note: declared here
  291 |     QFILE_MAYBE_NODISCARD bool open(OpenMode flags) override;
      |                                ^~~~
[272/558] Building CXX object launcher/CMakeFiles/Launcher_logic.dir/modplatform/helpers/OverrideUtils.cpp.o
.../PrismLauncher/launcher/modplatform/helpers/OverrideUtils.cpp: In function ‘void Override::createOverrides(const QString&, const QString&, const QString&)’:
.../PrismLauncher/launcher/modplatform/helpers/OverrideUtils.cpp:18:14: warning: ignoring return value of ‘virtual bool QFile::open(QIODeviceBase::OpenMode)’, declared with attribute ‘nodiscard’ [-Wunused-result]
   18 |     file.open(QFile::WriteOnly);
      |     ~~~~~~~~~^~~~~~~~~~~~~~~~~~
In file included from /usr/include/qt6/QtCore/qdir.h:11,
                 from /usr/include/qt6/QtCore/qdiriterator.h:8,
                 from /usr/include/qt6/QtCore/QDirIterator:1,
                 from .../PrismLauncher/launcher/modplatform/helpers/OverrideUtils.cpp:3:
/usr/include/qt6/QtCore/qfile.h:291:32: note: declared here
  291 |     QFILE_MAYBE_NODISCARD bool open(OpenMode flags) override;
      |                                ^~~~
.../PrismLauncher/launcher/modplatform/helpers/OverrideUtils.cpp: In function ‘QStringList Override::readOverrides(const QString&, const QString&)’:
.../PrismLauncher/launcher/modplatform/helpers/OverrideUtils.cpp:46:14: warning: ignoring return value of ‘virtual bool QFile::open(QIODeviceBase::OpenMode)’, declared with attribute ‘nodiscard’ [-Wunused-result]
   46 |     file.open(QFile::ReadOnly);
      |     ~~~~~~~~~^~~~~~~~~~~~~~~~~
/usr/include/qt6/QtCore/qfile.h:291:32: note: declared here
  291 |     QFILE_MAYBE_NODISCARD bool open(OpenMode flags) override;
      |                                ^~~~
[282/558] Building CXX object launcher/CMakeFiles/Launcher_logic.dir/modplatform/legacy_ftb/PackInstallTask.cpp.o
.../PrismLauncher/launcher/modplatform/legacy_ftb/PackInstallTask.cpp: In member function ‘void LegacyFTB::PackInstallTask::install()’:
.../PrismLauncher/launcher/modplatform/legacy_ftb/PackInstallTask.cpp:150:22: warning: ignoring return value of ‘virtual bool QFile::open(QIODeviceBase::OpenMode)’, declared with attribute ‘nodiscard’ [-Wunused-result]
  150 |         packJson.open(QIODevice::ReadOnly | QIODevice::Text);
      |         ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/qt6/QtCore/qdir.h:11,
                 from /usr/include/qt6/QtCore/QtCore:91,
                 from /usr/include/qt6/QtNetwork/QtNetworkDepends:3,
                 from /usr/include/qt6/QtNetwork/QtNetwork:3,
                 from .../PrismLauncher/launcher/net/NetJob.h:39,
                 from .../PrismLauncher/launcher/meta/BaseEntity.h:22,
                 from .../PrismLauncher/launcher/meta/Index.h:20,
                 from .../PrismLauncher/launcher/modplatform/legacy_ftb/PackInstallTask.h:4,
                 from .../PrismLauncher/launcher/modplatform/legacy_ftb/PackInstallTask.cpp:36:
/usr/include/qt6/QtCore/qfile.h:291:32: note: declared here
  291 |     QFILE_MAYBE_NODISCARD bool open(OpenMode flags) override;
      |                                ^~~~
[321/558] Building CXX object launcher/CMakeFiles/Launcher_logic.dir/Application.cpp.o
.../PrismLauncher/launcher/Application.cpp: In lambda function:
.../PrismLauncher/launcher/Application.cpp:1971:18: warning: ignoring return value of ‘virtual bool QFile::open(QIODeviceBase::OpenMode)’, declared with attribute ‘nodiscard’ [-Wunused-result]
 1971 |         file.open(QIODevice::WriteOnly);
      |         ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/qt6/QtCore/qdir.h:11,
                 from /usr/include/qt6/QtCore/QDir:1,
                 from .../PrismLauncher/launcher/FileSystem.h:45,
                 from .../PrismLauncher/launcher/DataMigrationTask.h:7,
                 from .../PrismLauncher/launcher/Application.cpp:46:
/usr/include/qt6/QtCore/qfile.h:291:32: note: declared here
  291 |     QFILE_MAYBE_NODISCARD bool open(OpenMode flags) override;
      |                                ^~~~
[398/558] Building CXX object launcher/CMakeFiles/Launcher_logic.dir/ui/dialogs/AboutDialog.cpp.o
.../PrismLauncher/launcher/ui/dialogs/AboutDialog.cpp: In function ‘QString {anonymous}::getLicenseHtml()’:
.../PrismLauncher/launcher/ui/dialogs/AboutDialog.cpp:127:18: warning: ignoring return value of ‘virtual bool QFile::open(QIODeviceBase::OpenMode)’, declared with attribute ‘nodiscard’ [-Wunused-result]
  127 |     dataFile.open(QIODevice::ReadOnly);
      |     ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/qt6/QtCore/qdir.h:11,
                 from /usr/include/qt6/QtCore/QtCore:91,
                 from /usr/include/qt6/QtNetwork/QtNetworkDepends:3,
                 from /usr/include/qt6/QtNetwork/QtNetwork:3,
                 from .../PrismLauncher/launcher/net/NetJob.h:39,
                 from .../PrismLauncher/launcher/ui/dialogs/AboutDialog.cpp:44:
/usr/include/qt6/QtCore/qfile.h:291:32: note: declared here
  291 |     QFILE_MAYBE_NODISCARD bool open(OpenMode flags) override;
      |                                ^~~~
[509/558] Building CXX object tests/CMakeFiles/MojangVersionFormat.dir/MojangVersionFormat_test.cpp.o
.../PrismLauncher/tests/MojangVersionFormat_test.cpp: In static member function ‘static QJsonDocument MojangVersionFormatTest::readJson(QString)’:
.../PrismLauncher/tests/MojangVersionFormat_test.cpp:12:22: warning: ignoring return value of ‘virtual bool QFile::open(QIODeviceBase::OpenMode)’, declared with attribute ‘nodiscard’ [-Wunused-result]
   12 |         jsonFile.open(QIODevice::ReadOnly);
      |         ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/qt6/QtCore/qdir.h:11,
                 from /usr/include/qt6/QtCore/qtemporarydir.h:9,
                 from /usr/include/qt6/QtTest/qtestcase.h:15,
                 from /usr/include/qt6/QtTest/qtest.h:13,
                 from /usr/include/qt6/QtTest/QTest:1,
                 from .../PrismLauncher/tests/MojangVersionFormat_test.cpp:2:
/usr/include/qt6/QtCore/qfile.h:291:32: note: declared here
  291 |     QFILE_MAYBE_NODISCARD bool open(OpenMode flags) override;
      |                                ^~~~
.../PrismLauncher/tests/MojangVersionFormat_test.cpp: In static member function ‘static void MojangVersionFormatTest::writeJson(const char*, QJsonDocument)’:
.../PrismLauncher/tests/MojangVersionFormat_test.cpp:20:22: warning: ignoring return value of ‘virtual bool QFile::open(QIODeviceBase::OpenMode)’, declared with attribute ‘nodiscard’ [-Wunused-result]
   20 |         jsonFile.open(QIODevice::WriteOnly | QIODevice::Text);
      |         ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/qt6/QtCore/qfile.h:291:32: note: declared here
  291 |     QFILE_MAYBE_NODISCARD bool open(OpenMode flags) override;
      |                                ^~~~
At global scope:
[519/558] Building CXX object tests/CMakeFiles/Library.dir/Library_test.cpp.o
.../PrismLauncher/tests/Library_test.cpp: In member function ‘LibraryPtr LibraryTest::readMojangJson(QString)’:
.../PrismLauncher/tests/Library_test.cpp:51:22: warning: ignoring return value of ‘virtual bool QFile::open(QIODeviceBase::OpenMode)’, declared with attribute ‘nodiscard’ [-Wunused-result]
   51 |         jsonFile.open(QIODevice::ReadOnly);
      |         ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/qt6/QtCore/qdir.h:11,
                 from /usr/include/qt6/QtCore/qtemporarydir.h:9,
                 from /usr/include/qt6/QtTest/qtestcase.h:15,
                 from /usr/include/qt6/QtTest/qtest.h:13,
                 from /usr/include/qt6/QtTest/QTest:1,
                 from .../PrismLauncher/tests/Library_test.cpp:36:
/usr/include/qt6/QtCore/qfile.h:291:32: note: declared here
  291 |     QFILE_MAYBE_NODISCARD bool open(OpenMode flags) override;
      |                                ^~~~
At global scope:
[532/558] Building CXX object tests/CMakeFiles/Version.dir/Version_test.cpp.o
.../PrismLauncher/tests/Version_test.cpp: In member function ‘void VersionTest::test_flexVerTestVector_data()’:
.../PrismLauncher/tests/Version_test.cpp:109:25: warning: ignoring return value of ‘virtual bool QFile::open(QIODeviceBase::OpenMode)’, declared with attribute ‘nodiscard’ [-Wunused-result]
  109 |         vector_file.open(QFile::OpenModeFlag::ReadOnly);
      |         ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/qt6/QtCore/qdir.h:11,
                 from /usr/include/qt6/QtCore/qtemporarydir.h:9,
                 from /usr/include/qt6/QtTest/qtestcase.h:15,
                 from /usr/include/qt6/QtTest/qtest.h:13,
                 from /usr/include/qt6/QtTest/QTest:1,
                 from .../PrismLauncher/tests/Version_test.cpp:16:
/usr/include/qt6/QtCore/qfile.h:291:32: note: declared here
  291 |     QFILE_MAYBE_NODISCARD bool open(OpenMode flags) override;
      |                                ^~~~

```

### This issue is unique

- [x] I have searched the issue tracker and did not find an issue describing my bug.

--- a/launcher/Application.cpp
+++ b/launcher/Application.cpp
@@ -1968,7 +1968,9 @@ bool Application::handleDataMigration(const QString& currentData,
 
     auto setDoNotMigrate = [&nomigratePath] {
         QFile file(nomigratePath);
-        file.open(QIODevice::WriteOnly);
+        if (!file.open(QIODevice::WriteOnly)) {
+            qWarning() << "setDoNotMigrate failed; Failed to open file '" << file.fileName() << "' for writing!";
+        }
     };
 
     // create no-migrate file if user doesn't want to migrate

--- a/launcher/FileSystem.cpp
+++ b/launcher/FileSystem.cpp
@@ -1002,7 +1002,10 @@ QString createShortcut(QString destination, QString target, QStringList args, QS
     if (!destination.endsWith(".desktop"))  // in case of isFlatpak destination is already populated
         destination += ".desktop";
     QFile f(destination);
-    f.open(QIODevice::WriteOnly | QIODevice::Text);
+    if (!f.open(QIODevice::WriteOnly | QIODevice::Text)) {
+        qWarning() << "Failed to open file '" << f.fileName() << "' for writing!";
+        return QString();
+    }
     QTextStream stream(&f);
 
     auto argstring = quoteArgs(args, "'", "'\\''");

--- a/launcher/modplatform/helpers/OverrideUtils.cpp
+++ b/launcher/modplatform/helpers/OverrideUtils.cpp
@@ -15,7 +15,10 @@ void createOverrides(const QString& name, const QString& parent_folder, const QS
     FS::ensureFilePathExists(file_path);
 
     QFile file(file_path);
-    file.open(QFile::WriteOnly);
+    if (!file.open(QFile::WriteOnly)) {
+        qWarning() << "Failed to open file '" << file.fileName() << "' for writing!";
+        return;
+    }
 
     QDirIterator override_iterator(override_path, QDirIterator::Subdirectories);
     while (override_iterator.hasNext()) {
@@ -43,7 +46,10 @@ QStringList readOverrides(const QString& name, const QString& parent_folder)
 
     QStringList previous_overrides;
 
-    file.open(QFile::ReadOnly);
+    if (!file.open(QFile::ReadOnly)) {
+        qWarning() << "Failed to open file '" << file.fileName() << "' for reading!";
+        return previous_overrides;
+    }
 
     QString entry;
     do {

--- a/launcher/modplatform/legacy_ftb/PackInstallTask.cpp
+++ b/launcher/modplatform/legacy_ftb/PackInstallTask.cpp
@@ -153,25 +153,29 @@ void PackInstallTask::install()
     QFile packJson(m_stagingPath + "/minecraft/pack.json");
     QDir jarmodDir = QDir(m_stagingPath + "/unzip/instMods");
     if (packJson.exists()) {
-        packJson.open(QIODevice::ReadOnly | QIODevice::Text);
-        QJsonDocument doc = QJsonDocument::fromJson(packJson.readAll());
-        packJson.close();
+        if (packJson.open(QIODevice::ReadOnly | QIODevice::Text)) {
+            QJsonDocument doc = QJsonDocument::fromJson(packJson.readAll());
+            packJson.close();
 
-        // we only care about the libs
-        QJsonArray libs = doc.object().value("libraries").toArray();
+            // we only care about the libs
+            QJsonArray libs = doc.object().value("libraries").toArray();
 
-        for (const auto& value : libs) {
-            QString nameValue = value.toObject().value("name").toString();
-            if (!nameValue.startsWith("net.minecraftforge")) {
-                continue;
+            for (const auto& value : libs) {
+                QString nameValue = value.toObject().value("name").toString();
+                if (!nameValue.startsWith("net.minecraftforge")) {
+                    continue;
+                }
+
+                GradleSpecifier forgeVersion(nameValue);
+
+                components->setComponentVersion("net.minecraftforge",
+                                                forgeVersion.version().replace(m_pack.mcVersion, "").replace("-", ""));
+                packJson.remove();
+                fallback = false;
+                break;
             }
-
-            GradleSpecifier forgeVersion(nameValue);
-
-            components->setComponentVersion("net.minecraftforge", forgeVersion.version().replace(m_pack.mcVersion, "").replace("-", ""));
-            packJson.remove();
-            fallback = false;
-            break;
+        } else {
+            qWarning() << "Failed to open file '" << packJson.fileName() << "' for reading!";
         }
     }
 

--- a/launcher/net/HttpMetaCache.cpp
+++ b/launcher/net/HttpMetaCache.cpp
@@ -112,7 +112,10 @@ auto HttpMetaCache::resolveEntry(QString base, QString resource_path, QString ex
     qint64 file_last_changed = finfo.lastModified().toUTC().toMSecsSinceEpoch();
     if (file_last_changed != entry->m_local_changed_timestamp) {
         QFile input(real_path);
-        input.open(QIODevice::ReadOnly);
+        if (!input.open(QIODevice::ReadOnly)) {
+            qWarning() << "Failed to open file '" << input.fileName() << "' for reading!";
+            return staleEntry(base, resource_path);
+        }
         QString md5sum = QCryptographicHash::hash(input.readAll(), QCryptographicHash::Md5).toHex().constData();
         if (entry->m_md5sum != md5sum) {
             selected_base.entry_list.remove(resource_path);

--- a/launcher/ui/dialogs/AboutDialog.cpp
+++ b/launcher/ui/dialogs/AboutDialog.cpp
@@ -124,9 +124,14 @@ QString getCreditsHtml()
 QString getLicenseHtml()
 {
     QFile dataFile(":/documents/COPYING.md");
-    dataFile.open(QIODevice::ReadOnly);
-    QString output = markdownToHTML(dataFile.readAll());
-    return output;
+    if (dataFile.open(QIODevice::ReadOnly)) {
+        QString output = markdownToHTML(dataFile.readAll());
+        dataFile.close();
+        return output;
+    } else {
+        qWarning() << "Failed to open file '" << dataFile.fileName() << "' for reading!";
+        return QString();
+    }
 }
 
 }  // namespace

--- a/tests/Library_test.cpp
+++ b/tests/Library_test.cpp
@@ -48,7 +48,10 @@ class LibraryTest : public QObject {
     LibraryPtr readMojangJson(const QString path)
     {
         QFile jsonFile(path);
-        jsonFile.open(QIODevice::ReadOnly);
+        if (!jsonFile.open(QIODevice::ReadOnly)) {
+            qCritical() << "Failed to open file '" << jsonFile.fileName() << "' for reading!";
+            return LibraryPtr();
+        }
         auto data = jsonFile.readAll();
         jsonFile.close();
         ProblemContainer problems;

--- a/tests/MojangVersionFormat_test.cpp
+++ b/tests/MojangVersionFormat_test.cpp
@@ -9,7 +9,10 @@ class MojangVersionFormatTest : public QObject {
     static QJsonDocument readJson(const QString path)
     {
         QFile jsonFile(path);
-        jsonFile.open(QIODevice::ReadOnly);
+        if (!jsonFile.open(QIODevice::ReadOnly)) {
+            qWarning() << "Failed to open file '" << jsonFile.fileName() << "' for reading!";
+            return QJsonDocument();
+        }
         auto data = jsonFile.readAll();
         jsonFile.close();
         return QJsonDocument::fromJson(data);
@@ -17,7 +20,10 @@ class MojangVersionFormatTest : public QObject {
     static void writeJson(const char* file, QJsonDocument doc)
     {
         QFile jsonFile(file);
-        jsonFile.open(QIODevice::WriteOnly | QIODevice::Text);
+        if (!jsonFile.open(QIODevice::WriteOnly | QIODevice::Text)) {
+            qCritical() << "Failed to open file '" << jsonFile.fileName() << "' for writing!";
+            return;
+        }
         auto data = doc.toJson(QJsonDocument::Indented);
         qDebug() << QString::fromUtf8(data);
         jsonFile.write(data);

--- a/tests/Version_test.cpp
+++ b/tests/Version_test.cpp
@@ -106,7 +106,10 @@ class VersionTest : public QObject {
 
         QFile vector_file{ test_vector_dir.absoluteFilePath("test_vectors.txt") };
 
-        vector_file.open(QFile::OpenModeFlag::ReadOnly);
+        if (!vector_file.open(QFile::OpenModeFlag::ReadOnly)) {
+            qCritical() << "Failed to open file '" << vector_file.fileName() << "' for reading!";
+            return;
+        }
 
         int test_number = 0;
         const QString test_name_template{ "FlexVer test #%1 (%2)" };

